packages/@aws-cdk/aws-appsync/lib/private.ts

```diff
@@ -12,6 +15,76 @@ export enum SchemaMode {
   CODE = 'CODE',
 };
 
+/**
+ * Generates an addition to the schema
+ *
+ * ```
+ * prefix name interfaces directives {
+ *   field
+ *   field
+ *   ...
+ * }
+ * ```
+ *
+ * @param prefix the prefix for this additon (type, interface, enum, input, schema)
+ * @param name the name for this addition (some additions dont need this [i.e. schema])
+ * @param interfaceTypes the interface types if this is creating an object type
+ * @param directives the directives for this type
+ * @param fields the fields to reduce onto the addition
+ */
+export interface SchemaAdditionOptions {
+  readonly prefix: string;
+  readonly name?: string;
+  readonly interfaceTypes?: InterfaceType[];
+  readonly directives?: Directive[];
+  readonly fields: string[];
```

Comment:
can we move the `@param` descriptions down to the properties themselves?
`@param` is intended to be used for functions and this is an interface